src/WebJobs.Script.WebHost/Helpers/MediaTypeMap.cs

```diff
@@ -28,11 +28,11 @@ public MediaTypeHeaderValue GetMediaType(string fileExtension)
             }
 
             return _mediatypeMap.GetOrAdd(fileExtension,
-                (extension) =>
+                static (extension) =>
                 {
                     try
                     {
-                        if (_mimeMapping.TryGetContentType(fileExtension, out string mediaTypeValue))
+                        if (_mimeMapping.TryGetContentType(extension, out string mediaTypeValue))
```

Comment:
If you know that extensions is `".<something>"` then you can access `FileExtensionContentTypeProvider.Mapping` directly and it will prevent the internal substring or lookups on the argument since it expects a path.

Reply:
But apparently this is just to lazily create new `MediaTypeHeaderValue` instances based on the values from `FileExtensionContentTypeProvider`. In that case why not just eagerly create all the entries from the `Mapping` dictionary, and not use a `ConcurrentDictionary` at all, but a static `Dictionary` that already contains all the possible values.

Reply:
Another good question - I just did some benchmarking to compare here. Overall: the spin-up would be much slower (more so on a 2-core machine we're typically on). If we did the full spin-up we would indeed gain a bit of throughput, but at the expensive of startup time which is a major effort as well. Comparison of spin-up only:

|               Method |        Mean |       Error |    StdDev |  Gen 0 |  Gen 1 | Allocated |
|--------------------- |------------:|------------:|----------:|-------:|-------:|----------:|
|     ConcurrentSpinup |    407.1 ns |    296.8 ns |  16.27 ns | 0.1068 | 0.0010 |      2 KB |
| FullDictionarySpinup | 36,976.1 ns | 15,953.0 ns | 874.44 ns | 2.7466 | 0.4272 |     46 KB |

I think more importantly though: there may be a breaking change around Markdown - the `Mappings` version has `text/markdown` where we have `text/plain` here...not sure if that may subtle break someone but definitely possible. Here's the benchmark setup if anyone's curious: https://gist.github.com/NickCraver/e4515051e60d393ba4c124cb1526e660

Now that's not a blocker IMO, since we can simply overwrite it last thing to keep compat - just noting a practical difference from a full move.

I'll add this to the look-at list - my next 2 big things to look into are: 1) startup time, and 2) out-of-process workers that light up all the gRPC paths. 

Curious on thoughts here ^